cocotb/regression.py

```diff
@@ -152,35 +163,18 @@ def initialise(self):
                         _py_compat.raise_from(err, None)  # discard nested traceback
 
                     if not hasattr(_test, "im_test"):
-                        self.log.error("Requested %s from module %s isn't a cocotb.test decorated coroutine", test, module_name)
+                        self.log.error("Requested %s from module %s isn't a cocotb.test decorated coroutine",
+                                       test, module_name)
                         raise ImportError("Failed to find requested test %s" % test)
-                    self._queue.append(_test(self._dut))
-                    self.ntests += 1
+                    self.init_test(_test)
+
+                # only look in first module for all functions and don't complain if all functions are not found
                 break
 
+            # auto-discover
             for thing in vars(module).values():
                 if hasattr(thing, "im_test"):
-                    try:
-                        test = thing(self._dut)
-                        skip = test.skip
-                    except Exception:
-                        skip = True
-                        self.log.warning("Failed to initialize test %s" %
-                                         thing.name, exc_info=True)
-
-                    if skip:
-                        self.log.info("Skipping test %s" % thing.name)
-                        self.xunit.add_testcase(name=thing.name,
-                                                classname=module_name,
-                                                time="0.0",
-                                                sim_time_ns="0.0",
-                                                ratio_time="0.0")
-                        self.xunit.add_skipped()
-                        self.skipped += 1
-                        self._store_test_result(module_name, thing.name, None, 0.0, 0.0, 0.0)
-                    else:
-                        self._queue.append(test)
-                        self.ntests += 1
+                    self.init_test(thing)
```

Comment:
Let's make this `_init_test` to make it super clear that it's only to be used in this file.